### DISCUSSION TITLE
Multi-label diceloss

### DISCUSSION THREAD

**Discussion by AugustKRZhu** [Project-MONAI/MONAI#2758] [Q&A]
Hi there,
Thanks to the MONAI team.
Recently, the multilabel segment on CT has been studied,  where label 2 is a subset of label 1, and label 1 is the constrain of label 2. My question is whether Diceloss or other loss functions can calculate independent label1 (label1+overlap of label1 and label2) and label2.
![multilables](https://user-images.githubusercontent.com/34640207/128896703-ef797857-2c63-4753-85e7-d9fe75c2a1fb.jpg)

**Comment by Nic-Ma** [comment]
Hi @AugustKRZhu ,

Thanks for your interest here.
I think `DiceLoss` can work well for your case, you can refer to the Brats segmentation tutorial:
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/brats_segmentation_3d.ipynb
It's multi-labels task.

Thanks.

**Comment by AugustKRZhu** [comment]
Hi Nic @Nic-Ma ,

Thanks for your reply, I refer to the Brats segmentation tutorial and train my dataset now,  but I am confused about whether the dice output is correct. as shown in fig, the dice losses are not mean values, are they suitable?
![diceloss](https://user-images.githubusercontent.com/34640207/129060231-13599342-99cb-4e21-9ae4-258885f6000d.jpg)

thanks.

**Comment by Nic-Ma** [comment]
Hi @AugustKRZhu ,

Actually, it's the mean value by `reduction` in the `aggregate()`:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/meandice.py#L96

Thanks.

**Comment by AugustKRZhu** [comment]
Awesome, you are right, I found that if I try to average them, it will be a problem. Thanks. @Nic-Ma
